src/Extension/Extension.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Extension;
+
+use Twig\Environment;
+use Twig\Extension\Attribute\AsTwigFilter;
+use Twig\Extension\Attribute\AsTwigFunction;
+use Twig\Extension\Attribute\AsTwigTest;
+use Twig\TwigFilter;
+use Twig\TwigFunction;
+use Twig\TwigTest;
+
+/**
+ * Abstract class for extension using the new PHP 8 attributes to define filters, functions, and tests.
+ *
+ * @author Jérôme Tamarelle <jerome@tamarelle.net>
+ */
+abstract class Extension extends AbstractExtension
+{
+    public function getFilters(): \Generator
+    {
+        $reflectionClass = new \ReflectionClass($this);
+        foreach ($reflectionClass->getMethods() as $method) {
+            foreach ($method->getAttributes(AsTwigFilter::class) as $attribute) {
+                $attribute = $attribute->newInstance();
+                $options = $attribute->options;
+                if (!\array_key_exists('needs_environment', $options)) {
+                    $param = $method->getParameters()[0] ?? null;
+                    $options['needs_environment'] = $param && 'env' === $param->getName() && Environment::class === $param->getType()->getName();
+                }
+                $firstParam = $options['needs_environment'] ? 1 : 0;
+                if (!\array_key_exists('needs_context', $options)) {
+                    $param = $method->getParameters()[$firstParam] ?? null;
+                    $options['needs_context'] = $param && 'context' === $param->getName() && 'array' === $param->getType()->getName();
+                }
+                $firstParam += $options['needs_context'] ? 1 : 0;
+                if (!\array_key_exists('is_variadic', $options)) {
+                    $param = $method->getParameters()[$firstParam] ?? null;
+                    $options['is_variadic'] = $param && $param->isVariadic();
+                }
+
+                yield new TwigFilter($attribute->name ?? $method->getName(), [$this, $method->getName()], $options);
```

Comment:
Should the default name be converted from camelCase to snake_case, to match the Twig convention? This makes it more difficult to find the declaration of a twig function/filter/test in PHP source code. But it does ensure that developers follow the naming convention by default.

Reply:
Given that Twig function names must be globally unique while method names are unique only to the class, I'm not even sure we want this automatic name. I'd rather make the name mandatory in the attribute.

Thus, this name is a public API of the Twig extension, so it means that any guessing logic we introduce is then set in stone as any change to it as a BC break.